gpii/node_modules/processReporter/src/processesBridge.js

```diff
@@ -0,0 +1,329 @@
+/*!
+GPII Process Reporter processes bridge -- gpii.processes.
+
+Copyright 2014 Inclusive Design Research Centre, OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+/*global require */
+
+"use strict";
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+gpii.processes = fluid.registerNamespace("gpii.processes");
+
+fluid.defaults("gpii.processes", {
+    gradeNames: ["gpii.processes.native", "fluid.component"],
+    events: {
+        onRunStateChange: null,
+        onStateChange: null
+    },
+    invokers: {
+        findSolutionsByCommands: {
+            funcName: "gpii.processes.findSolutionsByCommands",
+            args: ["{arguments}.0"]   // array of command names
+        },
+        findSolutionsByPids: {
+            funcName: "gpii.processes.findSolutionsByPids",
+            args: ["{arguments}.0"]   // array of pids (process ids)
+        },
+        findProcessByPid: {
+            funcName: "gpii.processes.findProcessByPid",
+            args: ["{arguments}.0", "{arguments}.1"] // pid, procArray (optional)
+        },
+        findProcessesByCommand: {
+            funcName: "gpii.processes.findProcessesByCommand",
+            args: ["{arguments}.0", "{arguments}.1"]
+            // command, procArray (optional)
+        },
+        findFirstProcessByCommand: {
+            funcName: "gpii.processes.findFirstProcessByCommand",
+            args: ["{arguments}.0", "{arguments}.1"]
+            // command, procArray (optional)
+        },
+        hasStateChanged: {
+            funcName: "gpii.processes.hasStateChanged",
+            args: ["{arguments}.0"]   // process info structure
+        },
+        isRunning: {
+            funcName: "gpii.processes.isRunning",
+            args: ["{arguments}.0"]   // state (string)
+        },
+        hasSwitchRunState: {
+            funcName: "gpii.processes.hasSwitchedRunState",
+            args: ["{arguments}.0"]   // monitor info structure
+        },
+        updateProcInfo: {
+            funcName: "gpii.processes.updateProcInfo",
+            args: ["{arguments}.0"]   // process info structure
+        },
+        initProcInfoNotRunning: {
+            funcName: "gpii.processes.initProcInfoNotRunning",
+            args: ["{arguments}.0"]   // command name (string)
+        },
+        initMonitor: {
```

Comment:
@kaspermarkus To be pedantic:  you want all of the "find the solution/process and tell me its state" functionality, but to mothball, "track the state of the solution/process"?

Okay.